src/map/pc.cpp

```diff
@@ -11956,15 +11957,17 @@ bool pc_equipitem(map_session_data *sd,short n,int req_pos,bool equipswitch)
 			for( i = 0; i < MAX_SLOTS; i++ ) {
 				if (!sd->inventory.u.items_inventory[n].card[i])
 					continue;
-
+				current_equip_card_id = sd->inventory.u.items_inventory[n].card[i];
 				std::shared_ptr<item_data> data = item_db.find(sd->inventory.u.items_inventory[n].card[i]);
 
 				if ( data != nullptr ) {
 					if (data->equip_script && (pc_has_permission(sd,PC_PERM_USE_ALL_EQUIPMENT) || !itemdb_isNoEquip(data.get(), sd->bl.m)))
 						run_script(data->equip_script,0,sd->bl.id,fake_nd->bl.id);
```

Comment:
I would prefer to have the current_equip_card_id just around this run_script call.